lambdas/s3select/index.py

```diff
@@ -0,0 +1,60 @@
+"""
+"""
+import os
+from urllib.parse import urlencode
+
+from botocore.auth import SigV4Auth
+from botocore.awsrequest import AWSRequest
+from botocore.session import Session
+
+import requests
+
+from t4_lambda_shared.decorator import api
+from t4_lambda_shared.utils import get_default_origins
+
+SERVICE = 's3'
+REGION = os.environ['AWS_REGION']
+
+REQUEST_HEADERS_TO_FORWARD = {'content-type', 'x-amz-content-sha256', 'x-amz-user-agent'}
+REQUEST_HEADERS_TO_SIGN = {'host', 'x-amz-content-sha256', 'x-amz-user-agent'}
+RESPONSE_HEADERS_TO_FORWARD = {'content-type'}
+
+credentials = Session().get_credentials()
+auth = SigV4Auth(credentials, SERVICE, REGION)
+session = requests.Session()
+
+
+@api(cors_origins=get_default_origins())
+def lambda_handler(request):
+    """
+    """
+    bucket, key = request.pathParameters['proxy'].split('/', 1)
+    host = f'{bucket}.s3.amazonaws.com'
+
+    url = f'https://{host}/{key}?{urlencode(request.args)}'
+
+    headers = { k: v for k, v in request.headers.items() if k in REQUEST_HEADERS_TO_FORWARD }
+    headers['host'] = host
+
+    aws_request = AWSRequest(
+        method=request.method,
```

Comment:
we need `POST` for s3 select, but it can also be used to write to buckets. the role should really guarantee that write is not permitted, but this still makes me nervous. consider scoping PUT requests to the `?select=`?

Reply:
Well, actually, it is kinda scary, and CORS doesn't guarantee anything. I'll check for `?select`.